### PR TITLE
Adds clipping of data via clip_interval argument

### PR DESCRIPTION
**Pull request by jeffreypaul15**
Similar to how it's done in sunpy, `plot_map()` accepts `clip_interval` as an argument :
```py
plotter.plot_map(m, clip_interval=(1, 99)*u.percent)
```
The above plot produces : 
image
as opposed to no clip interval : 
image

`vmin` and `vmax` is calculated based on the normalized mesh data, should we also allow accepting of `vmin` and `vmax` as separate arguments?

### CODE REVIEW

**Comment by nabobalis** [comment]
> `vmin` and `vmax` is calculated based on the normalized mesh data, should we also allow accepting of `vmin` and `vmax` as separate arguments?

How much effort would it be?

**Comment by jeffreypaul15** [comment]
> > `vmin` and `vmax` is calculated based on the normalized mesh data, should we also allow accepting of `vmin` and `vmax` as separate arguments?
> 
> How much effort would it be?

Just a few lines of code, is this functionality required?

**Comment by nabobalis** [comment]
> > > `vmin` and `vmax` is calculated based on the normalized mesh data, should we also allow accepting of `vmin` and `vmax` as separate arguments?
> > 
> > 
> > How much effort would it be?
> 
> Just a few lines of code, is this functionality required?

I can see people asking for a vmin,vmax but we can add it at a later date if we want to.

**Comment by jeffreypaul15** [comment]
> > > > `vmin` and `vmax` is calculated based on the normalized mesh data, should we also allow accepting of `vmin` and `vmax` as separate arguments?
> > > 
> > > 
> > > How much effort would it be?
> > 
> > 
> > Just a few lines of code, is this functionality required?
> 
> I can see people asking for a vmin,vmax but we can add it at a later date if we want to.

Sure

**Comment by nabobalis** [comment]
I think it's worth adding a basic unit test.

**Comment by jeffreypaul15** [comment]
> I think it's worth adding a basic unit test.

Yeah, I'll add one in

File: sunkit_pyvista/plotter.py [hunk 1]
```diff
@@ -119,7 +128,8 @@ def _pyvista_mesh(self, m):
         grid['data'] = m.plot_settings['norm'](data)
         return grid
 
-    def plot_map(self, m, **kwargs):
+    @u.quantity_input
+    def plot_map(self, m, clip_interval: u.percent = None, **kwargs):
```
Comment (dstansby) [inline sunkit_pyvista/plotter.py:132]: The new argument needs documenting below in the docstring.

**Review by dstansby — CHANGES_REQUESTED** [review]
Looks 👍  - needs a test, and a small addition to the docstring.

File: sunkit_pyvista/plotter.py [hunk 2]
```diff
@@ -119,20 +128,34 @@ def _pyvista_mesh(self, m):
...
         **kwargs :
             Keyword arguments are handed to `pyvista.Plotter.add_mesh`.
         """
         cmap = kwargs.pop('cmap', m.cmap)
         mesh = self._pyvista_mesh(m)
-        self.plotter.add_mesh(mesh, cmap=cmap, **kwargs)
+
+        if clip_interval is not None:
+            if len(clip_interval) == 2:
+                clim = self._get_clim(data=mesh['data'],
```
Comment (nabobalis) [inline sunkit_pyvista/plotter.py:151]: Codecov says this line isn't covered?

File: sunkit_pyvista/plotter.py [hunk 3]
```diff
@@ -119,20 +128,34 @@ def _pyvista_mesh(self, m):
...
+                                      clip_interval=clip_interval)
+            else:
+                raise ValueError("Clip percentile interval must be "
```
Comment (nabobalis) [inline sunkit_pyvista/plotter.py:154]: Codecov says this line isn't covered?

File: sunkit_pyvista/plotter.py (hunk 2 above)
Comment (jeffreypaul15) [reply sunkit_pyvista/plotter.py:151]: Ah, I forgot to add this in. I'll fix it

**Comment by dstansby** [comment]
Looks like this needs a rebase

**Review by dstansby — APPROVED** [review]

**Pull request merged**